platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1261,9 +1261,15 @@ - (id)mgl_jsonInterpolationExpressionObject {
         NSArray *controlPoints = [self.arguments[curveTypeIndex + 1].collection mgl_jsonExpressionObject];
         [interpolationArray addObjectsFromArray:controlPoints];
     }
+
+    NSDictionary<NSNumber *, NSExpression *> *stops = self.arguments[curveTypeIndex + 2].constantValue;
+
+    if (stops.count == 0) {
```

Comment:
This code assumes that all interpolations must have a stops dictionary. `MGLExpressionInterpolationModeLinear` may not have a stops dictionary.

Reply:
Linear interpolation still requires stops. Perhaps you’re thinking of the identity functions of yore.

Reply:
You are right. This is not true https://github.com/mapbox/mapbox-gl-native/pull/11723#discussion_r183526278